### ISSUE TITLE
Idea: create `step` method on `VirtualTimeScheduler`

### ISSUE THREAD

**Issue by alexreardon** [ReactiveX/rxjs#2300]
Currently Schedulers have a `.flush` method to flush its queue. However, for testing purposes it is really useful to just step through one action at a time.

Here is a little class I made that does the desired effect:

```js
import { VirtualTimeScheduler } from 'rxjs';

// An RxJS Scheduler that allows you to step through one
// async operation at a time.
export default class SteppableScheduler extends VirtualTimeScheduler {
    step() {
        const action = this.actions.shift();
        if (!action) {
            return;
        }

        action.execute(action.state, action.delay);
        this.frame = action.delay;
    }
}
```

I can post some example usage if needed

**Comment by mattpodwysocki** [comment]
@alexreardon yes, @trxcllnt and I are working on such a design to port over many of those features to the virtual time scheduler in vNext, so hopefully soon.  See https://github.com/ReactiveX/rxjs/issues/1873

**Comment by kwonoj** [comment]
And if you need v4 compatible interfaces allowing manual time advancing, https://github.com/kwonoj/rxjs-testscheduler-compat is available as immediate workaround.

**Comment by alexreardon** [comment]
It looks like that other issue covers this one and more. I am happy to have this one closed in favour of #1873 . 

In the meantime i'll continue to extend `VirtualTimeScheduler` with a little method as it is working really well without brining in another dependency.

Cheers!

**Comment by blesh** [comment]
Closed in favor of #1873
